src/rdkafka.h

```diff
@@ -673,6 +673,27 @@ rd_kafka_resp_err_t rd_kafka_fatal_error (rd_kafka_t *rk,
                                           char *errstr, size_t errstr_size);
 
 
+/**
+ * @brief Trigger a fatal error for testing purposes.
```

Comment:
ideally I think this functionality should be tested via integration tests using a mock broker, or intercepting and modifying calls to a real broker - I don't like the idea of having methods in a public API for testing purposes. I admit though that is difficult to set up and that the testing in this PR brings value. In the event we start doing integration testing as described above though, I think we should get rid of this method. so could maybe mark as such, but just my opinion.

Reply:
As stated elsewhere, it is practically impossible to trigger idempotent producer fatal errors without a full mock broker.

I think this API serves an important purpose for our users to allow them to easily test their (error handling) code, we wouldn't want to require them setting up a eos-breaking mock broker just to verify their code logic.

Reply:
very good point that users would find this useful for testing. ok, sounds good to me.